icaruscode/Decode/ChannelMapping/ChannelMapPostGres_tool.cc

```diff
@@ -165,23 +133,24 @@ ChannelMapPostGres::ChannelMapPostGres(fhicl::ParameterSet const &pset)
 bool ChannelMapPostGres::SelectPeriod(RunPeriod period) {
   
   auto const iPeriod = static_cast<unsigned int>(period);
-  TableNames_t const* newSet = &(TableNameSets.at(iPeriod));
+  std::string newPMTTimestamp = PMTTimestampSet.at(iPeriod);
   
-  if (fCurrentTable == newSet) {
+  if (fCurrentPMTTimestamp == newPMTTimestamp) {
     mf::LogDebug("ChannelMapPostGres")
       << "Period #" << iPeriod << " already selected";
   }
-  else if (fCurrentTable) {
+  else if ( !fCurrentPMTTimestamp.empty() ) {
     mf::LogDebug("ChannelMapPostGres") << "Switched from period #"
-      << (fCurrentTable - &(TableNameSets[0])) << " to #" << iPeriod;
+      << std::find(PMTTimestampSet.begin(), PMTTimestampSet.end(), fCurrentPMTTimestamp) 
```

Comment:
Doesn't
```suggestion
      << std::find(PMTTimestampSet.begin(), PMTTimestampSet.end(), fCurrentPMTTimestamp) 
```
return an iterator? what does this print?
I suggest that the period "time stamps" be printed instead of the number. The SQLite did not do that just because there was nothing easily available to print, while here we do have it.

Same comment for the previous branch if this `if`.

Reply:
I agree it makes much more sense to directly print the timestamp strings themselves. New commit deals with that.